spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDao.java

```diff
@@ -368,8 +368,9 @@ public TaskExecutionRowMapper() {
 		@Override
 		public TaskExecution mapRow(ResultSet rs, int rowNum) throws SQLException {
 			long  id = rs.getLong("TASK_EXECUTION_ID");
+			int exitCode = rs.getInt("EXIT_CODE");
 			return new TaskExecution(id,
-					(Integer) rs.getObject("EXIT_CODE"),
+					!rs.wasNull() ? exitCode : null,
```

Comment:
Since there is an explicit coupling between lines 371 and 373 (if you were to reorder 370 and 371, it would break), can we factor those into a method that just returns the appropriate value?